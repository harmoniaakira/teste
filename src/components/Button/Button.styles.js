import styled from 'styled-components'

export const ButtonComponent = styled.button`
  width: 280px;
  min-height: 48px;
  font-size: 12px;
  line-height: 24px;
  font-weight: 700;
  border: none;
  border-radius: 100px;
  box-shadow: 0px 6px 10px rgba(0, 0, 0, 0.11);
  transition: background-color 0.5s ease;
  text-transform: uppercase;
  cursor: pointer;

  ${props => props.fullScreen 
    ? "margin: 0 auto;"
    : ""
  }

  ${props => props.disabled 
    ? "background-color: #E0E0E0;" +
      "color: #8E8E93;" +
      "cursor: default;"
    : "background-color: #054375;" +
      "color: white;"
  }

  &:hover {
    ${props => props.disabled 
      ? "background-color: #E0E0E0;"
      : "background: #043C64;"
    }
  }

  @media (min-width: 1440px) {
    width: 158px;
    float: right;
  }
`
export const ButtonPopup = styled.button`
  width: 280px;
  height: 48px;
  font-size: 14px;
  line-height: 16px;
  font-weight: 600;
  letter-spacing: 0.75px;
  border: none;
  border-radius: 100px;
  text-transform: uppercase;
  transition: background-color 0.5s ease;
  cursor: pointer;
  background-color: transparent;

  ${props => props.disabled 
    ? "color: #8E8E93;" +
      "cursor: default;"
    : "color: #054375;"
  }

  &:hover {
    ${props => props.disabled 
      ? "background-color: transparent;"
      : "background: rgba(5, 67, 117, .1);"
    }
  }

  @media (min-width: 1440px) {
    width: 158px;
    float: right;
  }
`