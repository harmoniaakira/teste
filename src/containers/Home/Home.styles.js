import styled from 'styled-components'

export const Content = styled.div`
  width: 518px;
  min-height: 170px;
  max-height: 928px;
  margin: 0;
  position: absolute;
  top: 50%;
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
`

export const Avatar = styled.img`

`