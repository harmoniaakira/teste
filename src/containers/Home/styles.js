import styled from 'styled-components'
import { Typography } from '@material-ui/core'

export const Container = styled.div`
  background: ${({ theme }) => theme.palette.primary.background};
  height: 100vh;
  text-align: center;
`

export const Title = styled(Typography)`
  margin: 0;
  margin-bottom: 10px;
`
