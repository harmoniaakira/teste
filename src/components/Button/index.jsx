import React from 'react'
import PropTypes from 'prop-types';
import { ButtonComponent, ButtonPopup } from './Button.styles'

const Button = ( { handleClick, disabled, children, popup } ) => {
    
	return ( popup 
		? 	<ButtonPopup 
				onClick={handleClick}
				disabled={disabled}
			>
				{children}
			</ButtonPopup>
		:	<ButtonComponent 
				onClick={handleClick}
				disabled={disabled}
			>
				{children}
			</ButtonComponent>
	)
}

Button.propTypes = {
	handleClick: PropTypes.func.isRequired,
	children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired,
	disabled: PropTypes.bool,
	popup: PropTypes.bool
	
}

Button.defaultProps = {
	disabled: false,
	popup: false
}

export default Button
