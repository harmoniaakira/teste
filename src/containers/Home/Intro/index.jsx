import avatarPic from 'assets/images/PictureView/avatar.png'
import step1 from 'assets/images/PictureView/step1.svg'
import step2 from 'assets/images/PictureView/step2.svg'
import step3 from 'assets/images/PictureView/step3.svg'
import AppHeader from 'components/AppHeader'
import Button from 'components/Button'
import Dialog from 'components/Dialog'
import { MobileView } from 'components/Layout/Layout.styles'
import Steps from 'components/Steps'
import Web from 'components/Web'
import { useSendPhotoScreen } from 'contexts/SendPhotoContext'
import React, { useState } from 'react'
import {
    Avatar,
    Content,
    HeaderText,
    HeaderTitle,
    WrapperTakePicture
} from '../SendPhoto.styles'
import {
    SCREEN_SEND_PHOTO
  } from '../constants'

const steps = [
    {
        id: 1,
        stepIcon: step1,
        stepText: "Posicione o dispositivo na altura dos olhos."
    },
    {
        id: 2,
        stepIcon: step2,
        stepText: "Procure um local iluminado para garantir a qualidade da captura."
    },
    {
        id: 3,
        stepIcon: step3,
        stepText: "Retire qualquer tipo de acessório. Boné, óculos, etc."
    }
]

const SendPictureIntro = () => {

    const [open, setOpen] = useState(false);
    const { setScreen } = useSendPhotoScreen()

    const handleClick = () => {
        setOpen(true)
    }

    const handleClose = () => {
        setOpen(false)
    }

    const submit = () => {
        setScreen(SCREEN_SEND_PHOTO)
    }

    return (
        <>
            <MobileView>
                <AppHeader avatar={avatarPic} avatarFullSize />
                <HeaderTitle>Vamos tirar uma foto?</HeaderTitle>
                <HeaderText>
                    Precisamos verificar a identidade do cliente! Siga as instruções abaixo ao preencher
                    o formulário:
                </HeaderText>
                <Steps steps={steps} />

                <div style={{ paddingBottom: 60 }}>
                    <Button handleClick={handleClick}>Continuar</Button>
                </div>
            </MobileView>

            <Web>
                <Content>
                    <WrapperTakePicture>
                        <Avatar src={avatarPic} />
                        <HeaderTitle>Validação do código</HeaderTitle>
                        <HeaderText>
                            Enviaremos um código de segurança para o cliente. Preencha seguindo as
                            intruções:
                        </HeaderText>

                        <Steps steps={steps} />
                    </WrapperTakePicture>

                    <Button handleClick={handleClick}>Continuar</Button>
                </Content>
            </Web>
            <Dialog
                open={open}
                title="Atenção"
                confirmButtonLabel="aceitar"
                cancelButtonLabel="voltar"
                onConfirm={submit}
                onCancel={handleClose}
                handleClose={handleClose}
            >
                Será necessário aceitar o acesso à câmera do dispositivo para realizarmos a
                próxima etapa.
            </Dialog>
        </>
    )
}

export default SendPictureIntro