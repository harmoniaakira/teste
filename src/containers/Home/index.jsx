import avatarMobile from 'assets/images/HomeView/avatar-mobile1.svg'
import avatarWeb from 'assets/images/HomeView/avatar-web1.svg'
import AppHeader from 'components/AppHeader'
import Button from 'components/Button'
import Greetings from 'components/Greetings'
import { ContainerPage, MobileView } from 'components/Layout/Layout.styles'
import Steps from 'components/Steps'
import Web from 'components/Web'
import step1 from 'assets/images/HomeView/step1.svg'
import step2 from 'assets/images/HomeView/step2.svg'
import step3 from 'assets/images/HomeView/step3.svg'
import React from 'react'
import { useHistory } from 'react-router-dom'
import { Avatar, Content } from './Home.styles'

const steps = [
  {
    id: 1,
    stepIcon: step1,
    stepText: "Informe corretamente os dados"
  },
  {
    id: 2,
    stepIcon: step2,
    stepText: "Preencha de acordo com documentos válidos"
  },
  {
    id: 3,
    stepIcon: step3,
    stepText: "Escolha onde enviar o token de ativação"
  }
]

const Home = () => {
  const history = useHistory();

  const handleClick = () => {
    history.push('/onboarding-pf-frontend/geolocalizacao');
  }

  return (
    <ContainerPage>
      <MobileView $padding="0 0 40px 0">
        <AppHeader avatar={avatarMobile} />
        <Greetings />
        <Steps steps={steps} />
        <Button handleClick={handleClick}>Continuar</Button>
      </MobileView>

      <Web>
        <Content>
          <Avatar src={avatarWeb} />
          <Greetings />
          <Steps steps={steps} />
          <Button handleClick={handleClick}>Continuar</Button>
        </Content>
      </Web>
    </ContainerPage>
  )
}

export default Home;
