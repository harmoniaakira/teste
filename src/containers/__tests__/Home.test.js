import React from 'react'
import { shallow } from 'enzyme';
import Home from '../Home/index'
import Button from 'components/Button';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
      }),
}));

describe('Home', () => {

    const home = shallow(<Home />)

    it('Redirects to correct URL on click', () => {
        home.find(Button).at(0).props().handleClick();
        expect(mockHistoryPush).toHaveBeenCalledWith('/onboarding-pf-frontend/geolocalizacao');
    });

    it('should useContext mock and shallow render a div tag', () => {
        jest.spyOn(React, 'useContext').mockImplementation(() => ({
          name: 'this is a mock context return value'
        }));
    
        expect(home).toMatchSnapshot()
    })
})