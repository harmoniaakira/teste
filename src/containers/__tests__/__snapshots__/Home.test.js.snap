// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should useContext mock and shallow render a div tag 1`] = `
<styled.div>
  <styled.div
    $padding="0 0 40px 0"
  >
    <AppHeader
      avatar="SvgrURL"
    />
    <Greetings />
    <Memo(Steps) />
    <Button
      disabled={false}
      handleClick={[Function]}
      popup={false}
    >
      Continuar
    </Button>
  </styled.div>
  <Web>
    <styled.div>
      <styled.img
        src="SvgrURL"
      />
      <Greetings />
      <Memo(Steps) />
      <Button
        disabled={false}
        handleClick={[Function]}
        popup={false}
      >
        Continuar
      </Button>
    </styled.div>
  </Web>
</styled.div>
`;
